cli/flox-rust-sdk/src/models/floxmeta.rs

```diff
@@ -172,6 +178,42 @@ impl FloxMeta {
 
         Ok(FloxMeta { git })
     }
+
+    /// Prune the local branch for a deleted environment. If there are no more
+    /// local branches, then also prune the remote branch.
+    pub fn prune_branches(
+        &self,
+        pointer: &ManagedPointer,
+        dot_flox_path: impl AsRef<Path>,
+    ) -> Result<(), FloxMetaError> {
+        // The path may no longer exist.
+        let unchecked_path = CanonicalPath::new_unchecked(dot_flox_path);
```

Comment:
From sync review with @ysndr: move this up to the registry so that it's clearer that we're constructing unchecked CanonicalPaths from PathBufs that were previously canonicalised and existed. Changing the registry to store CanonicalPath would require implementing an unchecked deserialisation on all of them.

Reply:
Done (squashed)